tools/build.py

```diff
@@ -500,28 +500,30 @@ def _generate_build_tree(
             # if neither code signing development identity nor team id are provided, don't code sign
             cmake_args += ["-DCMAKE_XCODE_ATTRIBUTE_CODE_SIGNING_ALLOWED=NO"]
 
-        if args.ios:
+        if args.ios or args.apple_sysroot == "macosx":
```

Comment:
I don't quite understand what this condition means now. what if `args.apple_sysroot` is unset for a MacOS build as it was before?

Reply:
would adding an additional build.py script argument makes more sense here? 

initially I was trying to avoid creating a new argument for `macosx` so by inferring it from the apple_sysroot value. but it does seem to be an issue for the general MacOS build when apple_sysroot is not set.

Reply:
sorry actually maybe I could just use `arg.build_apple_framework` for these conditional apple cmake definitions.

Reply:
there's some iOS specific logic here too. maybe `args.build_apple_framework or args.ios`, similar to ORT's build.py?

https://github.com/microsoft/onnxruntime/blob/50cbcf95877b60795f32c4538611f9a119bb0291/tools/ci_build/build.py#L1293